modules/conversation.js

```diff
@@ -383,7 +383,7 @@ Conversation.prototype = {
   // XXX this logic is weird. Shouldn't we just compare a list of URLs?
   _selectionChanged: function _Conversation_selectionChanged () {
     let gFolderDisplay = topMail3Pane(this).gFolderDisplay;
-    let messageIds = [x.messageId for each ([, x] in Iterator(this._initialSet))];
+    let messageIds = [x.messageId for ([, x] of Iterator(this._initialSet))];
```

Comment:
ok I suspect you somehow stopped halfway in your rewrite, so I'll just leave it here, and check a little while later... we might want to check this in after the branching for 2.9, since this is such a huge change, I'm afraid some bugs will fall under the radar before the release... what do you think?